pkg/k8sops/k8sops.go

```diff
@@ -49,15 +50,17 @@ type K8sops struct {
 	Kclient    kubernetes.Interface
 	KubeExt    apiextensionsclient.Interface
 	MasterHost string
+	logger     *zap.Logger
 }
 
 // New creates a new instance of k8sops
-func New(kubeCfgFile, masterHost string) (*K8sops, error) {
+func New(logger *zap.Logger, kubeCfgFile, masterHost string) (*K8sops, error) {
 	crdClient, kubeClient, kubeExt, err := newKubeClient(kubeCfgFile)
 	if err != nil {
-		logrus.Fatalf("could not init Kubernetes client: %s", err)
+		logger.Fatal("could not init Kubernetes client", zap.Error(err))
```

Comment:
It's bad practice to fatalf in functions like this - makes it harder to understand exit/crash points in the code. Could you return an error for this case instead and then if you need to fatalf the program, do so in the top level func (main/Run/whatever)

Reply:
Done